packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -1605,3 +1613,668 @@ class _RestorableScrollOffset extends RestorableValue<double?> {
+      _verticalFallbackController = ScrollController();
+    }
+    if (widget.horizontalDetails.controller == null) {
+      _horizontalFallbackController = ScrollController();
+    }
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    _configuration = ScrollConfiguration.of(context);
+    super.didChangeDependencies();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(axisDirectionToAxis(widget.horizontalDetails.direction) == Axis.horizontal);
+    assert(axisDirectionToAxis(widget.verticalDetails.direction) == Axis.vertical);
+
+    Widget result = _VerticalOuterDimension(
+      key: _verticalOuterScrollableKey,
+      axisDirection: widget.verticalDetails.direction,
+      controller: widget.verticalDetails.controller ?? _verticalFallbackController!,
+      physics: widget.verticalDetails.physics,
+      clipBehavior: widget.verticalDetails.clipBehavior ?? Clip.hardEdge,
+      incrementCalculator: widget.incrementCalculator,
+      excludeFromSemantics: widget.excludeFromSemantics,
+      restorationId: widget.restorationId == null ? null : 'OuterTwoDimensionalScrollable',
+      dragStartBehavior: widget.dragStartBehavior,
+      viewportBuilder: (BuildContext context, ViewportOffset verticalOffset) {
+        return _HorizontalInnerDimension(
+          key: _horizontalInnerScrollableKey,
+          axisDirection: widget.horizontalDetails.direction,
+          controller: widget.horizontalDetails.controller ?? _horizontalFallbackController!,
+          physics: widget.horizontalDetails.physics,
+          clipBehavior: widget.horizontalDetails.clipBehavior ?? Clip.hardEdge,
+          incrementCalculator: widget.incrementCalculator,
+          excludeFromSemantics: widget.excludeFromSemantics,
+          restorationId: widget.restorationId == null ? null : 'InnerTwoDimensionalScrollable',
+          dragStartBehavior: widget.dragStartBehavior,
+          viewportBuilder: (BuildContext context, ViewportOffset horizontalOffset) {
+            return widget.viewportBuilder(context, verticalOffset, horizontalOffset);
+          },
+        );
+      },
+    );
+
+    // Build scrollbars for 2 dimensions instead of 1.
+    // TODO(Piinks): The right way is to create a dual constructor, https://github.com/flutter/flutter/issues/122348
+    result = _configuration.buildScrollbar(
+      context,
+      _configuration.buildScrollbar(
+        context,
+        result,
+        widget.horizontalDetails.copyWith(
+          controller: widget.horizontalDetails.controller ?? _horizontalFallbackController,
+          // Horizontal scrollbar matches vertical scrollbar (for now until dual available)
+          direction: widget.verticalDetails.direction,
+        ),
+      ),
+      widget.verticalDetails.copyWith(controller: widget.verticalDetails.controller ?? _verticalFallbackController),
+    );
+
+    if (widget.restorationId != null) {
+      result = RestorationScope(
```

Comment:
You can wrap the restorationscope unconditionally even if widget.restorationId is null.